src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
Many people still support IE 8, so people will be using `Symbol` polyfills for a very long time, including with jQuery v3.0.0.

I definitely understand that the test might be too large. I do believe though that jQuery should do as many checks as it feels comfortable doing to ensure that `Symbol`s are well-behaved in the engine.